pkg/crc/image/image.go

```diff
@@ -0,0 +1,134 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/code-ready/crc/pkg/extract"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/directory"
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type imageHandler struct {
+	imageURI string
+}
+
+func uri(preset crcpreset.Preset, registryURI string) string {
+	r := strings.SplitN(registryURI, "/", 2)
+	registry, orgName := r[0], r[1]
+	uri := url.URL{
+		Host: registry,
+		Path: orgName + "/" + getImageName(preset) + ":" + version.GetCRCVersion(),
+	}
+	return uri.String()
```

Comment:
@cfergeau `fmt.Sprintf("%s/%s:%s", constants.RegistryURI, getImageName(preset), version.GetCRCVersion()` will not provide the uri and `strings.SplitN` can't have more that 2 elements because we put limit to 2 here.

Reply:
but what is the point of splitting it since `uri.String()` will anyway concatenate it again? The interesting part of the URI construction is `orgName + "/" + getImageName(preset) + ":" + version.GetCRCVersion()` which we are already doing by hand. So why not just directly build the URI we want instead of hiding this behind `SplitN` followed by `String()` ?